src/System.Text.Json/src/System/Text/Json/Node/JsonString.cs

```diff
@@ -46,7 +46,7 @@ public sealed class JsonString : JsonNode, IEquatable<JsonString>
         /// <exception cref="ArgumentOutOfRangeException">
         ///   The date and time is outside the range of dates supported by the calendar used by the invariant culture.
         /// </exception>
-        public JsonString(DateTime value) => Value = value.ToString("s", CultureInfo.InvariantCulture);
+        public JsonString(DateTime value) => Value = JsonHelpers.FormatDateTime(value);
```

Comment:
Why is this an improvement?  If this is a different format than "s", that's functionally concerning.  If this is the same format as "s" but is somehow faster, can we instead optimize the performance of DateTime.ToString for "s" so that all other such consumers benefit?

Reply:
> Why is this an improvement? If this is a different format than "s", that's functionally concerning. If this is the same format as "s" but is somehow faster, can we instead optimize the performance of DateTime.ToString for "s" so that all other such consumers benefit?

The improvement here is the consolidation of the formats between `Json[Reader|Writer]` and `JsonString`. The former uses the `O` format (more on the details below), while the latter used `s` (which doesn't round-trip properly).

The reason this doesn't just use `ToString("o")` directly, is because the JSON writer will trim off some unnecessary details that are always included in the default format:

https://github.com/dotnet/corefx/blob/8bc0584c5d84eb83bf5901a7f3de83549e5518e1/src/System.Text.Json/src/System/Text/Json/Writer/JsonWriterHelper.Date.cs#L13-L21

I'm guessing this is done to get as compact round-trippable JSON as possible.

Reply:
Ah.  Ok, I remember that discussion.  We were discussing adding a new formatting option and this was the expedient compromise.

Reply:
For reference, here's the PR adding the code; https://github.com/dotnet/corefx/pull/35966